Lib/typing.py

```diff
@@ -727,7 +792,42 @@ def __init__(self, name, *constraints, bound=None,
             self.__module__ = def_mod
 
 
-class ParamSpec(_Final, _Immutable, _TypeVarLike, _root=True):
+class TypeVarTuple(_Final, _root=True):
+
+    def __init__(self, name):
+        self.__name__ = name
+        # All unpacked usages of this TypeVarTuple instance should have
+        # the same identity, so instantiate this here.
+        self._unpacked = _UnpackedTypeVarTuple(self)
+
+    def __iter__(self):
+        yield self._unpacked
+
+    def __repr__(self):
+        return self.__name__
+
+
+class _UnpackedTypeVarTuple:
+
+    def __init__(self, type_var_tuple: TypeVarTuple):
+        self._type_var_tuple = type_var_tuple
+
+    def __repr__(self):
+        return '*' + self._type_var_tuple.__name__
+
+
+class Unpack:
+
+    def __new__(cls, *args, **kwargs):
+        raise TypeError("Cannot instantiate Unpack")
+
+    def __class_getitem__(self, type_var_tuple: TypeVarTuple):
+        if not isinstance(type_var_tuple, TypeVarTuple):
+            raise TypeError("Parameter to Unpack must be a TypeVarTuple")
+        return type_var_tuple._unpacked
```

Comment:
```
        # All unpacked usages of this TypeVarTuple instance should have
        # the same identity, so instantiate this here.
```

This looks like the main reason we have a separate `_UnpackedTypeVarTuple` class. We want `TypeVarTuple("Ts") != TypeVarTuple("Ts")` but `Unpack[Ts] == Unpack[Ts]`.

I'd prefer to eliminate the confusing middleman class. Can we instead go with something like the following?

```suggestion
class TypeVarTuple:
  def __init__(self, name):
    self.name = name
    
  def __iter__(self):
    yield Unpack[self]
 
  def __repr__(self):
    return self.name

class Unpack:
  def __class_getitem__(self, type_var_tuple):
    return Unpack(type_var_tuple, is_direct=False)

  def __init__(self, type_var_tuple, is_direct=True):
    if is_direct:
      raise TypeError("Cannot instantiate Unpack directly")
    self.type_var_tuple = type_var_tuple

  def __eq__(self, other):
    return isinstance(other, Unpack) and self.type_var_tuple == other.type_var_tuple
    
  def __repr__(self):
    return '*' + repr(self.type_var_tuple)
  
```

REPL testing:

```python
>>> TypeVarTuple("Ts") == TypeVarTuple("Ts")
False
>>> Ts = TypeVarTuple("Ts")
>>> Unpack(Ts)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<string>", line 18, in __init__
TypeError: Cannot instantiate Unpack directly
>>> Unpack[Ts] == Unpack[Ts]
True
>>> Ts
Ts
>>> Unpack[Ts]
*Ts
>>> isinstance(Unpack[Ts], Unpack)
True

```